src/screens/SDKPayment/WebSDK.res

```diff
@@ -161,28 +161,28 @@ module CheckoutForm = {
     let handleSubmit = async () => {
       open LogicUtils
       try {
-        let confirmParams =
-          [
-            (
-              "confirmParams",
-              [
-                ("return_url", returnUrl->JSON.Encode.string),
-                ("redirect", "always"->JSON.Encode.string),
-              ]
-              ->Dict.fromArray
-              ->JSON.Encode.object,
-            ),
-          ]->getJsonFromArrayOfJson
-        let res = await hyper.confirmPayment(confirmParams)
-        let status = res->getDictFromJsonObject->getOptionString("status")
-        switch status {
-        | Some(str) =>
-          switch str {
-          | "failed" => setPaymentStatus(_ => FAILED("Failed"))
-          | "succeeded" => setPaymentStatus(_ => SUCCESS)
-          | _ => setPaymentStatus(_ => CUSTOMSTATE)
+        let confirmParamsToPass = {
+          "elements": elements,
+          "confirmParams": [
+            ("return_url", returnUrl->JSON.Encode.string),
+            ("redirect", "always"->JSON.Encode.string),
+          ]->getJsonFromArrayOfJson,
+        }
+        let res = await hyper.confirmPayment(confirmParamsToPass->Identity.genericTypeToJson)
+        let responseDict = res->getDictFromJsonObject
+        let errorDict = responseDict->getDictfromDict("error")
+        if errorDict->getOptionString("type") !== Some("validation_error") {
+          let status = responseDict->getOptionString("status")
+          switch status {
+          | Some(str) =>
+            switch str {
+            | "failed" => setPaymentStatus(_ => FAILED("Failed"))
+            | "succeeded" => setPaymentStatus(_ => SUCCESS)
+            | _ => setPaymentStatus(_ => CUSTOMSTATE)
+            }
+          | None => setPaymentStatus(_ => CUSTOMSTATE)
           }
-        | None => setPaymentStatus(_ => CUSTOMSTATE)
+          setClientSecret(_ => None)
```

Comment:
why its moved over here

Reply:
Because in case of validation error it was getting set to null. We don't need to set to null.

Reply:
at any point we need to regenerate new client secret if we get some error

Reply:
yes if
`if errorDict->getOptionString("type") !== Some("validation_error") {`

This comes just don't redirect or clear the client secret.

Reply:
can you add some description or screenshot

Reply:
Updated